hal/src/typelevel.rs

```diff
@@ -152,11 +152,7 @@
 //! parameters in place of struct fields to represent the unknown types.
 //!
 //! ```
-//! struct Container<A, B>
-//! where
-//!     A: EnumOne,
-//!     B: EnumTwo,
-//! {
+//! struct Container<A: EnumOne, B: EnumTwo> {
```

Comment:
Are this and the other changes to eliminate the `where` clauses personal preference, or suggested by a tool?  I feel the old way is more illustrative, and since they are in docstrings this change seems to be moving in the wrong direction.